pkg/virt-handler/vm.go

```diff
@@ -38,8 +40,29 @@ func NewVMController(lw cache.ListerWatcher, domainManager virtwrap.DomainManage
 				queue.AddRateLimited(key)
 				return true
 			}
-
 			vm = v1.NewVMReferenceFromName(name)
+
+			// If we don't have the VM in the cache, it could be that it is currently migrating to us
+			result := restClient.Get().Name(vm.GetObjectMeta().GetName()).Resource("vms").Namespace(kubeapi.NamespaceDefault).Do()
+			if result.Error() == nil {
+				// So the VM still seems to exist
+				fetchedVM, err := result.Get()
+				if err != nil {
+					// Since there was no fetch error, this should have worked, let's back off
+					queue.AddRateLimited(key)
+					return true
+				}
+				if fetchedVM.(*v1.VM).Status.MigrationNodeName == host {
```

Comment:
This is always the name of the node which kubernetes uses to identify it and reach the node. I think that should be the only thing we need here.